third_party/move/move-vm/runtime/src/storage/loader/lazy.rs

```diff
@@ -127,4 +179,81 @@ where
     }
 }
 
-impl<'a, T> Loader for LazyLoader<'a, T> where T: ModuleStorage {}
+impl<'a, T> InstantiatedFunctionLoader for LazyLoader<'a, T>
+where
+    T: ModuleStorage,
+{
+    fn load_instantiated_function(
+        &self,
+        // For lazy loading, not used!
+        _config: &LegacyLoaderConfig,
+        gas_meter: &mut impl DependencyGasMeter,
+        traversal_context: &mut TraversalContext,
+        module_id: &ModuleId,
+        function_name: &IdentStr,
+        ty_args: &[TypeTag],
+    ) -> VMResult<LoadedFunction> {
+        let (module, function) =
+            self.load_function_definition(gas_meter, traversal_context, module_id, function_name)?;
+
+        let ty_args = ty_args
+            .iter()
+            .map(|ty_arg| {
+                self.metered_load_type(gas_meter, traversal_context, ty_arg)
+                    .map_err(|e| e.finish(Location::Undefined))
+            })
+            .collect::<VMResult<Vec<_>>>()
+            .map_err(|mut err| {
+                // User provided type argument failed to load. Set extra sub status to distinguish
+                // from internal type loading error.
+                if StatusCode::TYPE_RESOLUTION_FAILURE == err.major_status() {
+                    err.set_sub_status(EUSER_TYPE_LOADING_FAILURE);
+                }
+                err
+            })?;
+
+        Type::verify_ty_arg_abilities(function.ty_param_abilities(), &ty_args)
+            .map_err(|e| e.finish(Location::Module(module_id.clone())))?;
+
+        Ok(LoadedFunction {
+            owner: LoadedFunctionOwner::Module(module),
+            ty_args,
+            function,
+        })
+    }
+}
+
+impl<'a, T> ClosureLoader for LazyLoader<'a, T>
+where
+    T: ModuleStorage,
+{
+    fn load_closure(
+        &self,
+        gas_meter: &mut impl DependencyGasMeter,
+        traversal_context: &mut TraversalContext,
+        module_id: &ModuleId,
+        function_name: &IdentStr,
+        ty_args: &[TypeTag],
+    ) -> PartialVMResult<Rc<LoadedFunction>> {
+        self.load_instantiated_function(
+            // Dummy config: will not be used.
+            &LegacyLoaderConfig::noop(),
+            gas_meter,
+            traversal_context,
+            module_id,
+            function_name,
+            ty_args,
+        )
+        .map_err(|err| err.to_partial())
+        .map(Rc::new)
+    }
+}
+
+impl<'a, T> Loader for LazyLoader<'a, T>
+where
+    T: ModuleStorage,
+{
+    fn unmetered_module_storage(&self) -> &dyn ModuleStorage {
+        self.module_storage
```

Comment:
there's not many caller of this, I'm hoping we can get rid of this

Reply:
It is mostly a hack to get around not being able to pass `impl Loader` to native context, which can only take `dyn` (otherwise we need to refactor natives), so eventually we can get rid of this I hope - essentially same problem as ther was for `GasMeter` before